src/Shelly/Base.hs

```diff
@@ -306,6 +307,9 @@ echo_n     msg = traceEcho msg >> liftIO (TIO.putStr msg >> hFlush stdout)
 echo_err   msg = traceEcho msg >> liftIO (TIO.hPutStrLn stderr msg >> hFlush stdout)
 echo_n_err msg = traceEcho msg >> liftIO (TIO.hPutStr stderr msg >> hFlush stderr)
 
+echoWith :: (Text -> IO ()) -> Text -> Sh ()
```

Comment:
In case this is exported, it needs a `@since` annotation.

Reply:
Appended since